### DISCUSSION TITLE
srsUE ZeroMQ and srsRAN stuck on RRC Connected

### DISCUSSION THREAD

**Discussion by HesselTjeerdsma** [srsran/srsRAN_Project#120] [General Help] [answered]
Hello,

I am trying to create a setup using Open5Gs and srsRAN using [this](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup) tutorial. I am using the exact config files referenced in that tutorial. Next to that, I am running Open5Gs using cpf/upf split (confirmed working using UERANSIM). Next to that, I am using the newest commit from the srsRAN_4g branch (fa56836b1). srsue and srsran (gnb) are running on the same VM. The ngap pcap files have been included, as well as the gnb.log. 

What is going wrong here? I am complete stuck... Thanks for helping!

[pcap.zip](https://github.com/srsran/srsRAN_Project/files/12001657/pcap.zip)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/12001655/gnb.log)

**Comment by brendan-mcauliffe** [comment]
Hi @HesselTjeerdsma, 

Can you please attach the UE `mac_nr` pcaps and the UE logs? Also, it is much easier to read the logs when they are in `info` mode. Setting the logs to `debug` results in a lot of information being stored in the logs, which can make them hard to read quickly. Sending updated gNB logs in `info` mode would be a help.

**Comment by HesselTjeerdsma** [comment]
Hi @brendan-mcauliffe, 

Thanks for your quick reply! All info logs and PCAPs are in the attached zip file. Next to that, I forgot to mention that in my Open5Gs core we use 1 or 2 as tac (instead of 7 mentioned in the tutorial). Based on the tac, we select a certain upf. Let me know if you need more information. 
[debug_files.zip](https://github.com/srsran/srsRAN_Project/files/12003707/debug_files.zip)

I have also attached a screenshot of the subscriber as configured in Open5Gs and both the gnb and ue config files for clarity.  
<img width="825" alt="image" src="https://github.com/srsran/srsRAN_Project/assets/11350641/46004fa9-81f1-4f17-b225-8d5c2ce76308">
[gnb.yaml (formatted as txt)](https://github.com/srsran/srsRAN_Project/files/12003742/gnb_zmq.yaml.txt)
[ue.conf (formatted as txt)](https://github.com/srsran/srsRAN_Project/files/12003757/zmg.conf.txt)

**Comment by herlesupreeth** [comment]
I see in the UE log the following. @andrepuschmann looks like a bug in handling 5GS tracking area identity list.

@HesselTjeerdsma Can you please share your AMF configuration file so that we can reproduce this issue?

> 2023-07-10T15:23:00.544389 [ASN1   ] [E] [    0] Decoding Failed (5GS tracking area identity list): Length (10) does not equal expected length 7
> 2023-07-10T15:23:00.544391 [ASN1   ] [E] [    0] [/root/srsRAN_4G/lib/src/asn1/nas_5g_msg.cc][524] Decoding failure.
> 2023-07-10T15:23:00.544391 [ASN1   ] [E] [    0] [/root/srsRAN_4G/lib/src/asn1/nas_5g_msg.cc][3929] Decoding failure.
> 2023-07-10T15:23:00.544392 [ASN1   ] [E] [    0] [/root/srsRAN_4G/lib/src/asn1/nas_5g_msg.cc][3905] Decoding failure.
> 2023-07-10T15:23:00.544392 [NAS5G  ] [E] Unable to unpack complete NAS pdu

**Comment by HesselTjeerdsma** [comment]
@herlesupreeth I have included the amf, smf and upf config files. Please note that we have two upfs, and based on the tac, one of the two upfs is selected. Thanks for the help!
[configs.zip](https://github.com/srsran/srsRAN_Project/files/12012348/configs.zip)

**Reply by HesselTjeerdsma** [reply, marked as answer]
Interesting, whenever I change line 36 of the amf.yaml from  `tac: [1, 2]` to `tac: 1` I get a succesful pdu session:

```
RRC Connected
PDU Session Establishment successful. IP: 10.45.0.2
RRC NR reconfiguration successful.
```

**Reply by andrepuschmann** [reply]
As @herlesupreeth pointed out this is very likely a bug in the UE's NAS code. Feel free to open an issue in the other repo and share the logs there for reference. But it's a low-prio issue to fix this for us. That being said, I think it's worth pointing out this limitation in the appnote so users are aware of it. Thanks @HesselTjeerdsma for reporting.

**Comment by bijanunl** [comment]
I have tried to use the ZeroMQ-based virtual radio support provided by both srsUE and srs gNB for implementing srsRAN via the link:
https://himanshup.hashnode.dev/5g-sa-lab-setup-using-srsran-open5gs.

According to the blog that I have followed, the output for Start gNB looks exactly the same as the implementation that I did.

However, the issue is created when starting the UE. The RRC is connected, but the RRC releases at that time; that's why the PDU session is not generated.

I have tried Ubuntu 22.04, Ubuntu 20, and a virtual machine. But it showed the same error.
![no pdu session generate issue](https://github.com/srsran/srsRAN_Project/assets/24872206/91958234-47ab-43bf-81b3-8c2068cb561f)

Can you please suggest how to resolve it?

**Reply by pgawlowicz** [reply]
Please follow our official tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

**Reply by MrPluto0** [reply]
https://github.com/srsran/srsRAN_4G/issues/1304#issuecomment-2069121036 
I meet the same problem. hope works for you.

**Reply by pgawlowicz** [reply]
Hi, if you are not starting any traffic, a gnb will quickly release a UE. Please put this section into your gnb config file to make the release timer longer.

```
cu_cp:
  inactivity_timer: 7200            # Sets the UE/PDU Session/DRB inactivity timer to 7200 seconds. Supported: [1 - 7200].
```
